include/mbedtls/cipher.h

```diff
@@ -33,7 +33,24 @@
 #include <stddef.h>
 #include "mbedtls/platform_util.h"
 
-#if defined(MBEDTLS_GCM_C) || defined(MBEDTLS_CCM_C) || defined(MBEDTLS_CHACHAPOLY_C)
+/* Support for GCM either through MbedTLS SW implementation or PSA */
+#if defined(MBEDTLS_GCM_C) || \
+    (defined(MBEDTLS_USE_PSA_CRYPTO) && defined(PSA_WANT_ALG_GCM))
+#define MBEDTLS_CIPHER_HAVE_GCM
```

Comment:
Hmm, I'm not entirely fond of this: it suggests you can use the normal cipher API and get GCM, but actually when it's provided only by a driver, this would only work if you use `cipher_setup_psa()` (which btw was deprecated some time ago)...

Reply:
Note: the commit message says internal symbols, but this is a public header.

Reply:
Mmm good point indeed. IIRC so far these changes to the `cipher` module in the PSA context are mostly used to get a list of supported/known algs+modes, not really to implement some action. 
Are you proposing to use a better naming here?

Reply:
I have trouble thinking of a name that would me me fully happy right now, but I'd be much less concerned about naming if we could keep the macro private.

Reply:
I think this should be achieved by [this commit](https://github.com/Mbed-TLS/mbedtls/pull/8297/commits/db1ca8fc3344df9c5b065ea0247d2df7a65ee539) since all the cipher's symbols are moved to `cipher_wrap.h`